packages/boilerplates/BoilerplatesLoader.ts

```diff
@@ -0,0 +1,184 @@
+    constructor(private _boilerplatesConfig: BoilerplatesConfig, private _folders: Folders, private _fileSystem: typeof FsExtra, private _logger: Logger) {
+        if (! this._fileSystem.existsSync(this._boilerplatesConfig.path)) {
+            this._boilerplatesConfig.store = this._boilerplatesConfig.store;
+        }
+    }
+    /**
+     * @inheritdoc
+     *
+     * @memberof BoilerplatesLoader
+     */
+    needsReload = true;
+    /**
+     * @inheritdoc
+     *
+     * @readonly
+     * @type {string}
+     * @memberof BoilerplatesLoader
+     */
+    get boilerplatesPath(): string {return this._boilerplatesConfig.path;}
+    /**
+     * @inheritdoc
+     *
+     * @readonly
+     * @type {BaseBoilerplate[]}
+     * @memberof BoilerplatesLoader
+     */
+    get loadedBoilerplates(): BaseBoilerplate[] {
+        if (!this._loadedBoilerplates || this.needsReload) return this.load();
+        return this._loadedBoilerplates;
+    }
+
+    /**
+     * @inheritdoc
+     * 
+     * @memberof BoilerplatesLoader
+     */
+    load(): BaseBoilerplate[] {
+        this._loadedBoilerplates = [];
+        let boilerplatesConfigObject: any = this._boilerplatesConfig.store;
+
+        Object.keys(boilerplatesConfigObject).forEach(key => {
+            let folderPath = path.resolve(boilerplatesConfigObject[key]);
+            if (!this._fileSystem.existsSync(folderPath)) {
+                this._logger.info(`Boilerplate path '${folderPath}' does not exist. Removing entry from boilerplates configuration`);
+                delete boilerplatesConfigObject[key];
+                this._boilerplatesConfig.store = boilerplatesConfigObject;
+            }
+            else this._loadedBoilerplates.push(this.readBoilerplateFromFolder(folderPath));
+        });
+        this.needsReload = false;
+        return this._loadedBoilerplates;
+    }
+    
+    private readBoilerplateFromFolder(folder: string): BaseBoilerplate {
+        let boilerplatePath = path.join(folder, 'boilerplate.json');
+        
+        if (!this._fileSystem.existsSync(boilerplatePath)) {
+            this._logger.info(`The path of a boilerplate defined in the boilerplates configuration does not exists. Path: ${boilerplatePath}`);
+            throw new Error(`Could not find boilerplate configuration in '${folder}'`);
+        }
+
+        let boilerplateObject = JSON.parse(this._fileSystem.readFileSync(boilerplatePath, 'utf8'));
+
+        return this.parseBaseBoilerplate(boilerplateObject, boilerplatePath);
+    }
+
+    private parseBaseBoilerplate(boilerplateObject: any, boilerplatePath: string): BaseBoilerplate {
+        if (boilerplateObject.type === artifactsBoilerplateType) 
+            return this.parseArtifactsBoilerplate(boilerplateObject, boilerplatePath);
+        else 
+            return this.parseBoilerplate(boilerplateObject, boilerplatePath);
+    }
+
+    private parseArtifactsBoilerplate(boilerplateObject: any, boilerplatePath: string): ArtifactsBoilerplate {
```

Comment:
It looks like you should split out into different parsers. This type is now breaking the open / closed principle. 
The concrete parsers would then also be responsible for parsing what is common - but they can delegate it to a thing that deals with the common properties.

`ICanParseBoilerplate`